src/main/java/team/unnamed/emojis/listener/ResourcePackApplyListener.java

```diff
@@ -40,15 +48,25 @@ public void onStatus(PlayerResourcePackStatusEvent event) {
             }
             case DECLINED: {
                 // TODO: De-hardcode the message
-                player.kickPlayer("§cPlease accept the resource pack");
+                if(plugin.getConfig().getBoolean("feature.require-pack"))
+                    player.kickPlayer(ChatColor.translateAlternateColorCodes('&', plugin.getConfig().getString("messages.kick")));
+                else
+                    player.sendMessage(ChatColor.translateAlternateColorCodes('&', plugin.getConfig().getString("messages.warn")));
                 break;
             }
             case FAILED_DOWNLOAD: {
                 Integer count = retries.get(player.getUniqueId());
                 if (count == null) {
                     count = 0;
                 } else if (count > 3) {
-                    player.kickPlayer("§cAn error occurred while downloading resource pack, please re-join");
+                    //player.kickPlayer("§cAn error occurred while downloading resource pack, please re-join");
+                    if(plugin.getConfig().getBoolean("feature.require-pack")){
+                        player.kickPlayer(ChatColor.translateAlternateColorCodes('&', plugin.getConfig().getString("messages.fail")));
+                    }
+                    else{
+                        player.sendMessage(ChatColor.translateAlternateColorCodes('&', plugin.getConfig().getString("messages.fail")));
+                        player.sendMessage(ChatColor.translateAlternateColorCodes('&', plugin.getConfig().getString("messages.warn")));
+                    }
```

Comment:
`player.sendMessage(ChatColor and more...)` creates a lot of boilerplate code. Can be simplified into a method `sendMessage(Player player, String messageNode)` with an internal translation of colors. And the same with `Player#kickPlayer`.

Reply:
Is there currently an internal implementation for translating colors?

Reply:
no, I meant to create a new method

Reply:
Why, they both would do the exact same thing and there is already a variable for the player?

Reply:
To be more clean like:
```java
public void onStatus(PlayerResourcePackStatusEvent event) {
    // All the code here...
    sendMessage(player, "message.a");
    sendMessage(player, "message.b");
}
```
with the method
```java
public void sendMessage(Player player, String messageNode) {
    String message = config.getString(messageNode);
    player.sendMessage(ChatColor.translateColorCode('&', message));
}
 ```